### PR TITLE
silx.gui: Check Qt/pyopengl platform compatibility; silx view/compare: Use matplotlib if OpenGL is not available

### LINKED ISSUE #3630: pyopengl 3.1.6 not supported
While trying the scatter 3D

```
$ mamba install pyopengl==3.1.6

$ python examples/plot3dUpdateScatterFromThread.py
```

Here is the result
```
Traceback (most recent call last):
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/silx/gui/_glutils/OpenGLWidget.py", line 223, in paintGL
    parent.paintGL()
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/silx/gui/plot3d/Plot3DWidget.py", line 350, in paintGL
    self._window.render(self.context(), self.getDevicePixelRatio())
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/silx/gui/plot3d/scene/window.py", line 364, in render
    self._renderWithOffscreenFramebuffer(context)
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/silx/gui/plot3d/scene/window.py", line 401, in _renderWithOffscreenFramebuffer
    wrap=gl.GL_CLAMP_TO_EDGE)
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/silx/gui/_glutils/FramebufferTexture.py", line 69, in __init__
    self._name = gl.glGenFramebuffers(1)
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/OpenGL/latebind.py", line 47, in __call__
    return self._finalCall( *args, **named )
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/OpenGL/wrapper.py", line 671, in wrapperCall
    result = wrappedOperation( *cArguments )
  File "/nobackup/lvalls1/valls/Software/miniconda3/envs/silxenv/lib/python3.7/site-packages/OpenGL/platform/baseplatform.py", line 425, in __call__
    self.__name__, self.__name__,
OpenGL.error.NullFunctionError: Attempt to call an undefined function glGenFramebuffers, check for bool(glGenFramebuffers) before calling

```

If i install `pyopengl 3.1.5`, everything works like a charm.

Any idea?

### PR DESCRIPTION
**Pull request by t20100**
This PR updates `isOpenGLAvailable` to check the compatibility between Qt and pyopengl selected platforms.
It also updates `silx view` and `silx compare` to fallback on matplotlib if OpenGL is selected but not available.

closes #3630

### CODE REVIEW

**Comment by vallsv** [comment]
I am not sure about your change with `options.use_opengl_plot`.

If you want to use opengl and it is not available, i prefer it to fail.
You still can remove if from the command line.

File: src/silx/app/compare/main.py [hunk 1]
```diff
@@ -70,9 +70,9 @@ def mainQt(options):
         logging.root.setLevel(logging.DEBUG)
 
     if options.use_opengl_plot:
-        backend = "gl"
+        backend = "opengl", "matplotlib"
```
Comment (vallsv) [inline src/silx/app/compare/main.py:73]: You can't specify 2 backends here right?

Comment (t20100) [reply src/silx/app/compare/main.py:73]: Yes you can!

But, I'll update it to fore OpenGL when requested from command line.

**Comment by t20100** [comment]
I reverted changes in `silx compare` and now `silx view --use-opengl-plot` forces OpenGL, but I left the matplotlib fallback when OpenGL is set in the settings.

**Pull request merged**